### ISSUE TITLE
Using manual Capybara sessions with poltergeist_billy?

### ISSUE THREAD

**Issue by destinf** [oesmith/puffing-billy#166]
Hi all!
I've been struggling to get poltergeist_billy working nicely with [manual capybara sessions](https://github.com/jnicklas/capybara#using-the-sessions-manually). When visiting a site with a Capybara session using the :poltergeist_billy driver, session.html returns an empty page, but the cache gets the correct content. Here's a stripped down version of my code:

```
require 'capybara'
require 'capybara/poltergeist'
require 'billy/rspec'

Billy.configure do |c|
  c.cache = true
  c.cache_request_headers = false
  c.ignore_params = ["http://www.google-analytics.com/__utm.gif",
                     "https://r.twimg.com/jot",
                     "http://p.twitter.com/t.gif",
                     "http://p.twitter.com/f.gif",
                     "http://www.facebook.com/plugins/like.php",
                     "https://www.facebook.com/dialog/oauth",
                     "http://cdn.api.twitter.com/1/urls/count.json"]
  c.path_blacklist = []
  c.merge_cached_responses_whitelist = []
  c.persist_cache = true
  c.ignore_cache_port = true # defaults to true
  c.non_successful_cache_disabled = false
  c.non_successful_error_level = :warn
  c.non_whitelisted_requests_disabled = false
  c.cache_path = 'spec/req_cache/'
  # c.proxy_host = 'example.com' # defaults to localhost
  # c.proxy_port = 12345 # defaults to random
  # c.proxied_request_host = nil
  # c.proxied_request_port = 80
  c.cache_request_body_methods = ['post', 'patch', 'put'] # defaults to ['post']
end

url = 'http://www.example.com'

session = Capybara::Session.new(:poltergeist_billy)
session.visit(url)

Billy.proxy.cache.instance_variable_get(:@cache).each_with_index do |(key,values), index|
  puts "CACHE_KEY #{index}: #{key}"
  puts "Response: #{values[:content]}"
end

puts "session.html: #{session.html}"
```

Would appreciate any debugging suggestions or ideas to try!

**Comment by ronwsmith** [comment]
Hmm, I haven't used the sessions manually before so I'm not sure how much I can help.  Try using selenium_billy so you can see what's going on perhaps?  Anything else available on the session after the visit?  What's supposed to be on that page?

Billy is only there to fetch additional assets requested from the page so it's odd that it would return anything for a blank page.

Does it work fine if you drop `_billy` from the session initialization?

**Comment by destinf** [comment]
It's been a while since I worked on this, but at the time working with just `:poltergeist` worked. With `:poltergeist_billy` the session only showed a blank html page. If I recall, it was something similar to `<html><body></body></html>`, regardless of url. I didn't get around to trying it on selenium_billy as I needed to get something out the door quickly.

That all being said I got back to this issue a few weeks ago, updated every gem related to capybara, poltergeist and puffing-billy to the latest and now the exact same code in the original post works as expected.
